app/contact/dns.py

```diff
@@ -0,0 +1,435 @@
+
+        self.transport.sendto(rdata, addr)
+
+    def datagram_received(self, data, addr):
+        """
+        Process a datagram received from the UDP socket
+
+        :param data: Raw data from the datagram endpoint
+        :type data: bytes
+        :param addr: Source address
+        :type addr: bytes
+        """
+        request = DNSRecord.parse(data)
+        asyncio.create_task(self.dns_work(request, addr))
+
+
+class C2Transmission(object):
+    """
+    C2Transmission
+    Maintains the state of an existing transmission communications.
+
+    :param id: Transmission ID
+    :type id: str
+    """
+
+    def __init__(self, id, paw, req_type):
+        """Constructor Method
+        """
+        self.id = id
+        self.data = dict()
+        self.paw = paw
+        self.req_type = req_type
+        self.final_contents = ""
+        self.response = None
+
+    def add_data(self, data, idx):
+        """
+        Adds data to an existing transmission if the indexed chunk does not already exist in the transmission.
+        """
+        if idx not in self.data:
+            self.data[idx] = data
+
+    def end(self, final_length):
+        """
+        Finalizes an existing transmission and concatenates all the data into the final contents.
+
+        Returns successfully if all sequential chunks up to `final_length` are present.
+        """
+        if all(k in self.data for k in range(final_length)):
+            for i in range(final_length):
+                self.final_contents += self.data[i]
+            return True
+        else:
+            return False
+
+
+class C2Resolver(BaseResolver):
+    def __init__(self, contact_svc, file_svc, suffix=''):
+        self.contact_svc = contact_svc
+        self.file_svc = file_svc
+        self.suffix = '54ndc47.%s' % suffix
+        self.transmissions = {}
+        self.cache = ExpiringDict(max_len=100000, max_age_seconds=100)
+        self.response_handler_lock = multiprocessing.Lock()
+        pass
+
+    async def handle_message(self, paw, req_type, data):
+        """
+        Handle requests depending on their request type.
+
+        :param req_type: Request type
+        :type req_type: int
+        :param data: Request data
+        :type data: dict
+        :return: Return dictionary from message processing function
+        :rtype: dict
+        """
+        data = json.loads(data)
+        ret = None
+        success = False
+        if req_type == 1:
+            success, ret = await self._get_instructions(data)
+        elif req_type == 2:
+            success, ret = await self._parse_results(data)
+        elif req_type == 3:
+            success, ret = await self._download_file(data)
+        elif req_type == 4:
+            success, ret = await self._upload_file(paw, data)
+        else:
+            success, ret = False, "Invalid request type"
+        return success, ret
+
+    async def resolve(self, request, handler):
+        """
```

Comment:
This function is way to large - let's start by breaking each of the ELIFs into individual functions.